src/olympia/lib/settings_base.py

```diff
@@ -1695,6 +1695,11 @@ def get_sentry_release():
     'send_default_pii': True,
 }
 
+
+# We need to load this before sentry_sdk.init or our reverse replacement is too late.
+from olympia.amo import urlresolvers  # noqa
```

Comment:
this could be at the top?  Don't know if that's helpful or not.

Reply:
I would do the monkeypatching at the top, yes. 

One thing that worries me though is importing something like that in a settings file - typically, settings limit the django-related imports they do to avoid accidentally triggering the initialization of something too early. It's difficult to see what consequences this has, but I feel it's a little bit risky.

Reply:
it's unfortunate, but we can't really avoid it unless we move the sentry initialization out of settings_base too

Reply:
we could split olympia.amo.urlresolvers into just `reverse` (and the supporting functions) and the other stuff too, to limit what's imported.

Reply:
Did we try loading sentry via the `AppConfig` when we integrated the sdk ? https://github.com/mozilla/addons-server/blob/master/src/olympia/core/apps.py

Reply:
no, sentry-sdk was just swapped in for raven so it was set-up in the same way.  (I don't know if raven supported loading via AppConfig as that was added before my time)

Reply:
their docs recommend initialization in settings https://docs.sentry.io/platforms/python/guides/django/

Reply:
Sentry docs and https://github.com/getsentry/sentry-python/issues/408#issuecomment-507667574 do seem to insist in doing sentry init in settings, so we probably want to keep it that way. Could you give the split into 2 different python modules a try ?

Reply:
https://github.com/mozilla/addons-server/pull/16624/commits/6ab82b0258772cf010135d12f2b6d7f64cfcf969 - this moves the functions (and refactors everywhere we import `reverse` because we were inconsistently importing it from `django.urls` in some places and `amo.urlresolvers` in others)